include/nbl/video/SPhysicalDeviceFeatures.h

```diff
@@ -0,0 +1,563 @@
+    //VkBool32           shaderUniformBufferArrayNonUniformIndexing;
+    //VkBool32           shaderSampledImageArrayNonUniformIndexing;
+    //VkBool32           shaderStorageBufferArrayNonUniformIndexing;
+    //VkBool32           shaderStorageImageArrayNonUniformIndexing;
+    //VkBool32           shaderInputAttachmentArrayNonUniformIndexing;
+    //VkBool32           shaderUniformTexelBufferArrayNonUniformIndexing;
+    //VkBool32           shaderStorageTexelBufferArrayNonUniformIndexing;
+    //VkBool32           descriptorBindingUniformBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingSampledImageUpdateAfterBind;
+    //VkBool32           descriptorBindingStorageImageUpdateAfterBind;
+    //VkBool32           descriptorBindingStorageBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingUniformTexelBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingStorageTexelBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingUpdateUnusedWhilePending;
+    //VkBool32           descriptorBindingPartiallyBound;
+    //VkBool32           descriptorBindingVariableDescriptorCount;
+    //VkBool32           runtimeDescriptorArray; // [FUTURE TODO]
+    
+    bool                 samplerFilterMinmax = false;   // ALIAS: VK_EXT_sampler_filter_minmax
+    
+    //VkBool32           scalarBlockLayout;     // or VK_EXT_scalar_block_layout // [FUTURE TODO]
+    
+    //VkBool32           imagelessFramebuffer;  // or VK_KHR_imageless_framebuffer 
+    
+    //VkBool32           uniformBufferStandardLayout;   // or VK_KHR_uniform_buffer_standard_layout
+    
+    // Don't expose:
+    //VkBool32           shaderSubgroupExtendedTypes;   // or VK_KHR_shader_subgroup_extended_types
+    
+    //VkBool32           separateDepthStencilLayouts;   // or VK_KHR_separate_depth_stencil_layouts
+    
+    // [TODO] And add implementation to engine
+    //VkBool32           hostQueryReset;                // or VK_EXT_host_query_reset
+    
+    //VkBool32           timelineSemaphore;             // or VK_KHR_timeline_semaphore
+    
+    // or VK_KHR_timeline_semaphore:
+    bool bufferDeviceAddress = false;
+    //VkBool32           bufferDeviceAddressCaptureReplay;
+    //VkBool32           bufferDeviceAddressMultiDevice;
+    
+    // or VK_KHR_vulkan_memory_model
+    //VkBool32           vulkanMemoryModel;
+    //VkBool32           vulkanMemoryModelDeviceScope;
+    //VkBool32           vulkanMemoryModelAvailabilityVisibilityChains;
+    
+    //VkBool32           shaderOutputViewportIndex;     // ALIAS: VK_EXT_shader_viewport_index_layer
+    //VkBool32           shaderOutputLayer;             // ALIAS: VK_EXT_shader_viewport_index_layer
+    //VkBool32           subgroupBroadcastDynamicId;    // if Vulkan 1.2 is supported
+
+
+
+
+    /* Vulkan 1.3 Core */
+
+    //VkBool32           robustImageAccess;                 //  or VK_EXT_image_robustness
```

Comment:
one of the EGL_robustness extensions and GL_robustness

Reply:
but maybe save yourself a headache and not expose any robustness right now, just note down what would have to be done.